pkg/sm/sm.go

```diff
@@ -139,6 +151,27 @@ func (smb *ServiceManagerBuilder) Build() *ServiceManager {
 	}
 }
 
+func initializeSecureStorage(secureStorage security.Storage)  error {
+	keyFetcher := secureStorage.Fetcher()
```

Comment:
If the key was deleted, the data would be lost I think. Reencrypting the data requires decrypting it in the first place, but if you don't have the key with which it was encrypted, you can't decrypt it.

Not sure what would be the benefit of generating it in the migration script. Also, this would couple the encryption key with a database and would not work with other types of storages